pkg/storage/utils/decomposedfs/node/permissions_unix.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+//go:build !darwin
+// +build !darwin
+
+package node
+
+import (
+	"syscall"
+
+	"github.com/pkg/xattr"
+)
+
+func isNoAttr(err error) bool {
+	if xerr, ok := err.(*xattr.Error); ok {
+		if serr, ok2 := xerr.Err.(syscall.Errno); ok2 {
+			return serr == syscall.ENODATA
```

Comment:
the function is called `isNoAttr` but the return checks `syscall.ENODATA`. Is your intent to have the function named `isNoData` or `isNoAttr`? The compiler does not yell because this file is ignored when building in darwin, but what was the intent?

Reply:
naming :-)

it was called `isNoData` 

## darwin
- `syscall.ENOATTR`

## other os
- `syscall.ENODATA`

The function name was only reflecting one case
BUT we need a function name which aplies for both


maybe `isNoDataOrNoAttr` ?

Reply:
I wouldn't bind the function name to its internal implementation, especially with shared cases like this. Plus `isNoAttr` is a little vague, especially being in the `node` package. Essentially the function is checking whether an error is of type `xattr.Error` regardless of the implementation. How about `attrNotSet` or to stick to the convention, `isAttrUnset` (and save mental load with a `not` on a func name), since this seems to be the underlying cause?

Reply:
furthermore, `ENOATTR` [is just an alias for `ENODATA`](https://github.com/pkg/xattr/blob/5e1af1232a565727d5eab3d201a62a360fd94949/xattr_darwin.go#L24-L27) 🙈

Reply:
ok, using now `isAttrUnset`